examples/conanfile/package_id/settings_and_options.rst

```diff
@@ -0,0 +1,206 @@
+                        os: Windows
+
+Whenever a value of the settings or subsettings changes, the package ID will be different to reflect that.
+
+The most common usage for settings is to model the different project-wide aspects that might influence the package ID.
+A recipe that does that will be:
+
+.. code-block:: python
+
+    from conan import ConanFile
+
+    class Pkg(ConanFile):
+        name = "pkg"
+        version = "1.0.0"
+        settings = "os", "arch", "compiler", "build_type"
+
+Now, compiling a package with different compiler versions will result into different package IDs:
+
+.. code-block:: bash
+   :emphasize-lines: 3, 7, 16, 27
+
+    $ conan create . --settings compiler.version=192
+    ...
+    pkg/1.0.0: Package '4f267380690f99b3ef385199826c268f63147457' created
+
+    $ conan create . --settings compiler.version=193
+    ...
+    pkg/1.0.0: Package 'c13a22a41ecd72caf9e556f68b406569547e0861' created
+
+    $ conan list pkg/1.0.0:*
+    Local Cache
+      pkg
+        pkg/1.0.0
+          revisions
+            f1f48830ecb04f3b328429b390fc5de8 (2024-03-07 09:21:07 UTC)
+              packages
+                4f267380690f99b3ef385199826c268f63147457
+                  info
+                    settings
+                      arch: x86_64
+                      build_type: Release
+                      compiler: msvc
+                      compiler.cppstd: 14
+                      compiler.runtime: dynamic
+                      compiler.runtime_type: Release
+                      compiler.version: 192
+                      os: Windows
+                c13a22a41ecd72caf9e556f68b406569547e0861
+                  info
+                    settings
+                      arch: x86_64
+                      build_type: Release
+                      compiler: msvc
+                      compiler.cppstd: 14
+                      compiler.runtime: dynamic
+                      compiler.runtime_type: Release
+                      compiler.version: 193
+                      os: Windows
+
+However, there are cases when a setting should not influence the resulting package ID.
+An example of this could be when a project generates an executable binary as a compilation result. In that case, the value of the compiler
+used is not relevant for the package and we can remove its influence in the package ID like so:
+
+.. code-block:: python
+
+    from conan import ConanFile
+
+    class Pkg(ConanFile):
+        name = "pkg"
+        version = "1.0.0"
+        settings = "os", "arch", "compiler", "build_type"
+
+        def build(self):
+            # self.settings.compiler value will be used here to compile
+
+        def package_id(self):
+            # Remove compiler setting from package ID
+            del self.info.settings.compiler
+
+Why not removing the setting from the `settings` attribute? Because the compiler value is still needed in the `build()` method to perform the compilation of the executable.
```

Comment:
I think this is at least controversial, and should be at least clarified, but probably changed. 

There is a much different use case if we are concerned about a ``tool_require``, like ``cmake/3.25``, or if we are developing our own final applications that is our product. For our own product I'd never recommend doing this, but this section is written in a way that it seems this must be done for all executables.

I'd suggest to elaborate it as two approaches:
- approach 1: removing the ``compiler``: why and when (tool-requires)
- approach 2: not removing the ``compiler``: why and when (my own apps)

And present the 2 approaches as equal alternatives

Reply:
Agree here - the setting should be in the `settings` attribute "if it is actually needed at any point". If it's not needed, I think the cleanest, easiest, way of handling it is just by not adding is an an attribute.

I would expand on this with actual examples - maybe add some CMake to the `build()` method or something (where the compiler setting is needed), and have another example of "building" a python only app (like meson) where the `compiler` setting is not needed in the slightest.

Reply:
I wasn't even covering the case where the ``compiler`` is not needed for building the package (python things), only the case where the ``compiler`` is really needed for the build, then the 2 approaches above. But it might be worth clarifiying the 3 different use cases, presenting them at the same level, not prioritizing the removal of settings in ``package_id()``.